presto-main/src/main/java/com/facebook/presto/execution/buffer/PageCodecMarker.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution.buffer;
+
+public enum PageCodecMarker
+{
+    COMPRESSED(1);
+
+    private final int mask;
+
+    PageCodecMarker(int flagBitPosition)
+    {
+        if (flagBitPosition <= 0 || flagBitPosition > 8) {
+            throw new IllegalArgumentException("PageCodecMarker flagBitPosition must be between 1 and 8. Found: " + flagBitPosition);
+        }
+        this.mask = (1 << (flagBitPosition - 1));
+    }
+
+    public boolean check(byte value)
+    {
+        return (Byte.toUnsignedInt(value) & mask) == mask;
+    }
+
+    public byte mark(byte value)
+    {
+        return (byte) (Byte.toUnsignedInt(value) | mask);
+    }
+
+    public byte clear(byte value)
+    {
+        return (byte) (Byte.toUnsignedInt(value) & (~mask));
+    }
+
+    public static byte none()
+    {
+        return 0;
+    }
+
+    public static byte create(PageCodecMarker marker)
```

Comment:
this method is a bit strange; it takes an object and returns a primitive type; the name create would suggest the opposite: take some primitives and construct an object; also, this method doesn't seem to be used in production code. Would you clarify its purpose?

Reply:
Sure, I'll add an comment block. Maybe the method would be better named `of` instead of `create`?